fdb-extensions/src/main/java/com/apple/foundationdb/async/hnsw/StorageTransform.java

```diff
@@ -65,17 +65,6 @@ public RealVector apply(@Nonnull final RealVector vector) {
     @Nonnull
     @Override
     public RealVector invertedApply(@Nonnull final RealVector vector) {
-        //
-        // Only transform the vector if it is needed. We make the decision based on whether the vector is encoded or
-        // not. When we switch on encoding, we apply the new coordinate system from that point onwards meaning that all
-        // vectors inserted before use the client coordinate system. For the inverted case, we only have to transform
-        // the encoded vectors as they are expressed in the internal coordinate system, while regular non-encoded
-        // vectors are already expressed in the client system.
-        //
-        if (!(vector instanceof EncodedRealVector)) {
```

Comment:
Wait what?

Reply:
Should we have more explicit tests about that transition? I'm glad to hear that one of the tests caught this, though it sounds like it's a bit brittle, in that parameters changing (e.g., deciding to change how much of a sample we need to enable RaBitQ) could cause us to lose coverage